src/porepy/numerics/ad/grid_operators.py

```diff
@@ -524,17 +520,31 @@ def __init__(
                 # The subdomain has no faces, so the projection does not exist.
                 mat_loc = sps.csr_matrix((0, tot_num_faces))
             mat.append(mat_loc)
-        self._projection: sps.spmatrix = sps.bmat([[m] for m in mat], format="csr")
+
+        self._projection: sps.spmatrix
         """Projection from subdomain faces to boundary grids cells."""
+        if len(mat) > 0:
+            self._projection = sps.bmat([[m] for m in mat], format="csr")
+        else:
+            self._projection = sps.csr_matrix((0, 0))
 
-    def subdomain_to_boundary(self) -> sps.spmatrix:
-        return self._projection
+    @property
+    def subdomain_to_boundary(self) -> Operator:
+        return SparseArray(self._projection, name="subdomains to boundaries projection")
 
-    def boundary_to_subdomain(self) -> sps.spmatrix:
-        return self._projection.transpose().tocsc()
+    @property
+    def boundary_to_subdomain(self) -> Operator:
+        return SparseArray(
+            self._projection.transpose().tocsc(),
+            name="boundaries to subdomains projection",
+        )
+    
+    # @property
+    # def dirichlet_to_subdomain(self) -> Operator:
+        # dirichlet = 
 
 
```

Comment:
```suggestion
```suggestion
    
    # @property
    # def dirichlet_to_subdomain(self) -> Operator:
        # dirichlet = 


```
```